src/Database/Connection.php

```diff
@@ -121,6 +122,19 @@ public function __construct($config)
     }
 
     /**
+     * Destructor
+     *
+     * Disconnects the driver to release the connection.
+     */
+    public function __destruct()
+    {
+        if ($this->_transactionStarted && class_exists('Cake\Log\Log')) {
+            Log::warning('The connection is going to be closed but the transaction is still active.');
+        }
```

Comment:
Recently this destructor was removed. Wouldn't that obviate the need for this change?